File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        hashmap = {}
        index = 1
        for val in numbers:
            if target - val in hashmap:
                return [hashmap[target-val],index]
            else:
                hashmap[val] = index
            index += 1

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int leftPtr = 0;
        int rightPtr = numbers.length - 1;

        while (leftPtr < rightPtr){

            if (numbers[leftPtr] + numbers[rightPtr] == target){
                return new int[]{leftPtr + 1,rightPtr + 1};
            }

            if (numbers[leftPtr] + numbers[rightPtr] > target)
            {
                rightPtr --;
            }
            else {
                leftPtr ++;
            }
        }
        return new int[]{}; 
    } 
}